src/iterator.rs

```diff
@@ -100,14 +156,32 @@ impl Iterator {
   }
 
   /// Get the left_span for the current position and move the cursor.
+  ///
+  /// ## Examples
+  /// ```rust
+  /// assert_eq!(flat_tree::iterator(0).left_span(), 0);
+  /// assert_eq!(flat_tree::iterator(1).left_span(), 0);
+  /// assert_eq!(flat_tree::iterator(3).left_span(), 0);
+  /// assert_eq!(flat_tree::iterator(23).left_span(), 16);
+  /// assert_eq!(flat_tree::iterator(27).left_span(), 24);
+  /// ```
   pub fn left_span(&mut self) -> usize {
-    self.index = self.index - self.factor / 2 + 1;
+    self.index = self.index + 1 - self.factor / 2;
```

Comment:
Actually found an issue here - unsigned integer underflow causes a panic (exposed by first 3 asserts). Moving `+1` prevents it.

Reply:
Ohh, great catch!